restore.py

```diff
@@ -59,6 +59,65 @@ def restoreFromBackup(backup, progress=lambda x: ()):
     except:
         pass
 
+    # should we restore old schema and move to MBR?
+    restore_partitions = None
+    if 'BOOT' not in backup_partition_layout and 'LOG' not in backup_partition_layout:
+        new_restore_partition = restore_partition
+        efi_boot = False
+
+        # remove unneeded partitions
+        if dsk.swap[0]:
+            tool.deletePartition(tool.partitionNumber(dsk.swap[1])) # swap
+        tool.deletePartitionIfPresent(boot_partnum) # boot
+
+        # use logs as root
+        root_partnum = tool.partitionNumber(dsk.root[1])
+        backup_partnum = tool.partitionNumber(backup_partition)
+        if dsk.logs[0]:
```

Comment:
The alignment of partitions may be fixed up during upgrade.

This doesn't restore the original (mis-)alignment during backup. I think that is fine, but it might be worth noting in a comment.

Reply:
Now that you mention not sure which alignment will the new backup have... surely the "old" logs will have 1MB alignment.

Reply:
Added a comment